File: scripts/package_release.py

import os
import zipfile
import shutil
from pathlib import Path

def create_dist_zip():
    print("📦 Packing ZenAI for Release...")
    
    # Define root
    ROOT_DIR = Path(os.getcwd())
    DIST_DIR = ROOT_DIR / "dist"
    DIST_DIR.mkdir(exist_ok=True)
    
    ZIP_NAME = DIST_DIR / "ZenAI_Dist.zip"
    
    # Define what to include
    INCLUDES = [
        "zena.py",
        "requirements.txt",
        "README.md",
        "USER_MANUAL.md",
        "config.json",
        "config.py",
        "utils.py",
        "config_system.py",
        "security.py",
        "async_backend.py",
        "locales",         # Folder
        "ui",              # Folder
        "zena_mode",       # Folder
        "ui_components.py",
        "settings.py",
        "state_management.py",
        "zena_startup.log", # Maybe not needed but okay
        "experimental_voice_lab", # Folder
        "tests"            # Include tests for verification
    ]
    
    # Create Zip
    with zipfile.ZipFile(ZIP_NAME, 'w', zipfile.ZIP_DEFLATED) as zipf:
        for item in INCLUDES:
            src_path = ROOT_DIR / item
            
            if not src_path.exists():
                print(f"⚠️ Warning: {item} not found, skipping.")
                continue
                
            if src_path.is_file():
                print(f"  Adding file: {item}")
                zipf.write(src_path, arcname=item)
                
            elif src_path.is_dir():
                print(f"  Adding dir:  {item}")
                for root, dirs, files in os.walk(src_path):
                    # Filter out garbage
                    if "__pycache__" in root: continue
                    if ".venv" in root: continue
                    if ".git" in root: continue
                    if "node_modules" in root: continue
                    
                    for file in files:
                        if file.endswith(".pyc"): continue
                        
                        full_path = Path(root) / file
                        rel_path = full_path.relative_to(ROOT_DIR)
                        zipf.write(full_path, arcname=str(rel_path))

    print(f"\n✅ Release created at: {ZIP_NAME}")
    print(f"📦 Size: {ZIP_NAME.stat().st_size / 1024 / 1024:.2f} MB")
    
    return ZIP_NAME

if __name__ == "__main__":
    create_dist_zip()

File: rust_output/src/package_release.rs
use anyhow::{Result, Context};
use std::fs::File;
use std::io::{self, Read, Write};
use std::path::PathBuf;

/// Helper: setup phase for create_dist_zip.
pub fn _do_create_dist_zip_setup() -> () {
    // Helper: setup phase for create_dist_zip.
    println!("{}", "📦 Packing ZenAI for Release...".to_string());
    let mut ROOT_DIR = PathBuf::from(std::env::current_dir().unwrap().to_str().unwrap().to_string());
    let mut DIST_DIR = (ROOT_DIR / "dist".to_string());
    DIST_DIR.create_dir_all();
    let mut ZIP_NAME = (DIST_DIR / "ZenAI_Dist.zip".to_string());
    let mut INCLUDES = vec!["zena.py".to_string(), "requirements.txt".to_string(), "README.md".to_string(), "USER_MANUAL.md".to_string(), "config::json".to_string(), "config::py".to_string(), "utils::py".to_string(), "config_system::py".to_string(), "security.py".to_string(), "async_backend::py".to_string(), "locales".to_string(), "ui".to_string(), "zena_mode".to_string(), "ui_components.py".to_string(), "settings::py".to_string(), "state_management::py".to_string(), "zena_startup.log".to_string(), "experimental_voice_lab".to_string(), "tests".to_string()];
    (INCLUDES, ROOT_DIR, ZIP_NAME)
}

/// Create dist zip.
pub fn create_dist_zip() -> Result<()> {
    // Create dist zip.
    let (mut INCLUDES, mut ROOT_DIR, mut ZIP_NAME) = _do_create_dist_zip_setup();
    let mut zipf = zipfile.ZipFile(ZIP_NAME, "w".to_string(), zipfile.ZIP_DEFLATED);
    {
        for item in INCLUDES.iter() {
            let mut src_path = (ROOT_DIR / item);
            if !src_path.exists() {
                println!("⚠️ Warning: {} not found, skipping.", item);
                continue;
            }
            if src_path.is_file() {
                println!("  Adding file: {}", item);
                zipf.write(src_path, /* arcname= */ item);
            } else if src_path.is_dir() {
                println!("  Adding dir:  {}", item);
                for (root, dirs, files) in os::walk(src_path).iter() {
                    if root.contains(&"__pycache__".to_string()) {
                        continue;
                    }
                    if root.contains(&".venv".to_string()) {
                        continue;
                    }
                    if root.contains(&".git".to_string()) {
                        continue;
                    }
                    if root.contains(&"node_modules".to_string()) {
                        continue;
                    }
                    for file in files.iter() {
                        if file.ends_with(&*".pyc".to_string()) {
                            continue;
                        }
                        let mut full_path = (PathBuf::from(root) / file);
                        let mut rel_path = full_path.relative_to(ROOT_DIR);
                        zipf.write(full_path, /* arcname= */ rel_path.to_string());
                    }
                }
            }
        }
    }
    println!("\n✅ Release created at: {}", ZIP_NAME);
    println!("📦 Size: {:.2} MB", ((ZIP_NAME.stat().st_size / 1024) / 1024));
    Ok(ZIP_NAME)
}
